ToDoApp/ToDoApp/SignUp/SignUpView.swift

```diff
@@ -0,0 +1,121 @@
+                            $0.clearButtonMode = .whileEditing
+                        })
+                    Divider()
+                    TextField("Email", text: $viewModel.email)
+                        .padding(.horizontal, 16.0)
+                        .keyboardType(.emailAddress)
+                        .textContentType(.emailAddress)
+                        .frame(minHeight: 60)
+                        .introspectTextField(customize: {
+                            $0.clearButtonMode = .whileEditing
+                        })
+                    Divider()
+                    SecureField("Password", text: $viewModel.password)
+                        .padding(.horizontal, 16.0)
+                        .textInputAutocapitalization(.never)
+                        .frame(minHeight: 60)
+                        .introspectTextField(customize: {
+                            $0.clearButtonMode = .whileEditing
+                        })
+                    Divider()
+                    HStack {
+                        Button(action: {
+                            viewModel.acceptTermsAndConditionsCheckBoxTapped()
+                        }, label: {
+                            Image(systemName: (viewModel.isAcceptTermsAndConditionsChecked ?  "checkmark.square.fill" : "squareshape"))
+                                .frame(minWidth: Constants.checkBoxSize, minHeight: Constants.checkBoxSize)
+                        }
+                        )
+                        Text("I agree with our Terms and Conditions")
+                        { string in
+                            string.foregroundColor = Color(.secondaryText)
+                            string.font = .caption
+                            if let range = string.range(of: "Terms") {
+                                string[range].foregroundColor = Color(.checkmarkButton)
+                            }
+                            if let range = string.range(of: "Conditions") {
+                                string[range].foregroundColor = Color(.checkmarkButton)
+                            }
+                        }
+                    }
+                    .padding(16)
+                }
+                Spacer()
+                Button("Create Account") {
+                    viewModel.signUpButtonTapped()
+                }
+                .foregroundColor(.white)
+                .buttonStyle(.bordered)
+                .frame(maxWidth: 343, maxHeight: 56)
+                .background(Color(.checkmarkButton))
+            }
+            Text("Already have an account? Sign In")
+            { string in
+                string.foregroundColor = Color(.secondaryText)
+                if let range = string.range(of: "Sign In") {
+                    string[range].foregroundColor = Color(.checkmarkButton)
+                }
+            }
+            .onTapGesture {
+                viewModel.signInButtonTapped()
+            }
```

Comment:
Button should be always considered before using onTapGesture. onTapGesture doesn't handle highlighting, it is not recognized as a button by the screen reader by default. I see no issues in using Button here
```
Button(action: {
    viewModel.signInButtonTapped()
}) {
    Text("Already have an account? Sign In")
             { string in
                 string.foregroundColor = Color(.secondaryText)
                 if let range = string.range(of: "Sign In") {
                     string[range].foregroundColor = Color(.checkmarkButton)
                 }
             }
}
```